src/modinv64_impl.h

```diff
@@ -626,4 +717,71 @@ static void secp256k1_modinv64_var(secp256k1_modinv64_signed62 *x, const secp256
     *x = d;
 }
 
+/* Do up to 25 iterations of 62 posdivsteps (up to 1550 steps; more is extremely rare) each until f=1.
+ * In VERIFY mode use a lower number of iterations (744, close to the median 756), so failure actually occurs. */
+#ifdef VERIFY
+#define JACOBI64_ITERATIONS 12
+#else
+#define JACOBI64_ITERATIONS 25
+#endif
```

Comment:
On the one hand, I really like this trick. But on the other hand, it creates a difference between VERIFY and non-VERIFY (and who knows, maybe this influences loop unrolling etc.)

Would it be a good idea to keep this 25 in the real code but create a test-only version of it (either in the tests file or here) that just runs with 12? Or maybe better, make the number of iterations an argument of the function? Since it's a "maybe" function, it should be correct for each iteration value.

Reply:
> I feel like VERIFY mode isn't to test the compiler's correctness, but to test the correctness of our source code.

That's the way I see it. too. [1] I think the fundamental problem here is that we don't really run tests in non-VERIFY mode. I'll comment in https://github.com/bitcoin-core/secp256k1/issues/1037, which is somewhat related.

 > A copy in tests could go out of sync,  [...]

Agreed, that's bad. 

> If we don't like it, I think the best alternative is just having maintainers occasionally manually setting the iteration count lower, and seeing if tests still pass.

That's even worse, we'll forget to run it.

> and having the iteration count be an argument still means somewhere higher in the stack there needs to be a decision what to set it to, which I think has the same problems.

> So... I'm not sure. I do see the uneasiness about it, but I feel that the current approach is really the only one that guarantees (a) it's actually exercised in realistic settings and (b) using the same code as is used in production. 

Hm, yeah, I don't think (a) is a big concern here: What this VERIFY tests is just handling the loop termination in case of too many iterations, and that's a few lines of code which doesn't care too much if the failure happened in an realistic input or not. So what I imagined it is that we run it for values of the iteration count in the test, and this will be good enough to exercise this code.

This would have the advantage that it minimizes the difference between VERIFY and non-VERIFY mode. But after thinking more about it, I agree that it's a somewhat moot goal. [2] Your comment about loop unrolling is right and we have a lot of code locations where we add some substantial amount of code in VERIFY (you introduce another one here in this PR, which I don't mind at all.)  

[1] Though to some extent, VERIFY can also help checking compiler, e.g., when a function is miscompiles and its result is VERIFIED somewhere else.
[2] I think my brain mixed up my thoughts with the discussion in https://github.com/bitcoin-core/secp256k1/pull/1169#pullrequestreview-1224682895,  but this was about SECP256K1_CHECKMEM_ENABLED being defined, not VERIFY.